examples/golang/src/example_with_zone/example_with_zone.go

```diff
@@ -18,103 +18,128 @@ package main
-	resp, err := http.Get("http://127.0.0.1:9191/getNewZone")
-	if err != nil {
-		panic("Error: getting new zone data (need start Acra with zones (-z) )")
-	}
-	defer resp.Body.Close()
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		panic(err)
-	}
-	zoneData := []byte(body)
-	//geting zone done
-	//	zone_data := []byte(`{"id":"ZXCxJAAWWbelaVCEcNp","public_key":"VUVDMgAAAC3zSak+Ah5wtcenUuD9PorpT8nmlecK2fG78nWsXZ9NEdotnH1B"}`)
-	//var parsed_zone_data map[string][]byte
-	var parsedZoneData ZoneData
-	err = json.Unmarshal(zoneData, &parsedZoneData)
-	if err != nil {
-		panic(err)
-	}
-	zonePublic := parsedZoneData.Public_key
-	zoneId := []byte(parsedZoneData.Id)
-
-	acrastruct, err := acrawriter.CreateAcrastruct(someData, &keys.PublicKey{Value: zonePublic}, zoneId)
-	if err != nil {
-		panic(err)
+	connectionString := fmt.Sprintf("user=%v password=%v dbname=%v host=%v port=%v", *user, *password, *dbname, *host, *port)
+	driver := "postgres"
+	if *mysql {
+		// username:password@protocol(address)/dbname?param=value
+		// https://github.com/go-sql-driver/mysql#dsn-data-source-name
+		connectionString = fmt.Sprintf("%v:%v@tcp(%v:%v)/%v", *user, *password, *host, *port, *dbname)
+		driver = "mysql"
 	}
 
-	db, err := sql.Open("postgres", CONNECTION_STRING)
+	db, err := sql.Open(driver, connectionString)
 	if err != nil {
 		log.Fatal(err)
 	}
+
 	err = db.Ping()
 	if err != nil {
 		log.Fatal(err)
 	}
-	fmt.Println("Create test2 table with command: 'CREATE TABLE IF NOT EXISTS test2(id INTEGER PRIMARY KEY, zone BYTEA, data BYTEA, raw_data TEXT);'")
-	_, err = db.Exec("CREATE TABLE IF NOT EXISTS test2(id INTEGER PRIMARY KEY, zone BYTEA, data BYTEA, raw_data TEXT);")
-	if err != nil {
-		panic(err)
+	if *mysql {
+		query := "CREATE TABLE IF NOT EXISTS test_example_with_zone(id INTEGER PRIMARY KEY, zone BINARY(24), data VARBINARY(1000), raw_data VARCHAR(1000));"
+		fmt.Printf("Create test_example_with_zone table with command: '%v'\n", query)
+		_, err = db.Exec(query)
+	} else {
+		query := "CREATE TABLE IF NOT EXISTS test_example_with_zone(id INTEGER PRIMARY KEY, zone BYTEA, data BYTEA, raw_data TEXT);"
+		fmt.Printf("Create test_example_with_zone table with command: '%v'\n", query)
+		_, err = db.Exec(query)
 	}
-
-	fmt.Println("Insert test data to table")
-	_, err = db.Exec("insert into test2 (id, zone, data, raw_data) values ($1, $2, $3, $4);", rand.Int31(), zoneId, acrastruct, string(someData))
 	if err != nil {
 		panic(err)
 	}
 
-	fmt.Println("Select from db with command: 'SELECT zone, data, raw_data FROM test2;'")
-	rows, err := db.Query(`SELECT zone, data, raw_data FROM test2;`)
-	defer rows.Close()
-	if err != nil {
-		log.Fatal(err)
-	}
-	var zone, data []byte
-	var rawData string
-	fmt.Println("zone, data - raw_data")
-	for rows.Next() {
-		err := rows.Scan(&zone, &data, &rawData)
+	if *data != "" {
+		rand.Seed(time.Now().UnixNano())
+		//get new zone over http
+		resp, err := http.Get("http://127.0.0.1:9191/getNewZone")
```

Comment:
is it the address of acraconnector? shall we put it into variable?

Reply:
yes. it's acraconnector's address and it used once, so I didn't assign it to variable and used default value